homeassistant/components/goodwe/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow to configure Goodwe inverters using their local API."""
+from __future__ import annotations
+
+import logging
+
+from goodwe import InverterError, connect
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_MODEL_FAMILY,
+    CONF_NETWORK_RETRIES,
+    CONF_NETWORK_TIMEOUT,
+    DEFAULT_NAME,
+    DEFAULT_NETWORK_RETRIES,
+    DEFAULT_NETWORK_TIMEOUT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Init object."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input: dict | None = None) -> FlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        settings_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
+                    default=self.config_entry.options.get(
+                        CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                    ),
+                ): int,
+                vol.Optional(
+                    CONF_NETWORK_RETRIES,
```

Comment:
Network retries and timeout should not be options but use sane defaults.

Reply:
We discussed this in the community prior to creating this PR and there was clear wish to have this possibility.
The PV inverters are deployed in very different (network) environments and their UDP protocol is notoriously unreliable.
Having the possibility to tweak the sane defaults to specific local conditions is for some users inevitable to reach reliable operation.

Reply:
Please give some different examples where defaults wouldn't be sufficient.

Reply:
The sane defaults are fine for maybe 90% of users, can't really measure that precisely, maybe even more. 
But there were several reports that users have unusually large request failure rate, usually due to UDP packet fragmentation/truncation. 
And no adjustment to WIFI or network infrastructure changed seemed to help, it's likely some design issue of the inverter HW.
And in such case, the only way to get any output is to decrease the poll rate and increase retries count (and sometimes also timeout).

Reply:
I suggest we continue without options and wait for specific reports and tests by concerned users and adjust the default at that time. We can increase timeout and retry at that time.

Reply:
I have asked the community to collect the non-default settings they use, already the first response was change of retry from default 3 to 10 (due to unstable net).
I can remove the timeout, that one can have sane default and/or be reasonable deducted from the retry count.

But I doubt there can be generic sane default for the retry count, since it heavily depends on the poll frequency.
If I use 1s polling, the retry should not be > 3, if I use 30s polling on unreliable net, than retry e.g. 15 is preferred.

Reply:
It would be better to allow the users that need a custom polling interval and retry to configure this in the automation that controls the polling. We can do this if we report the failed fetch accurately, eg by setting the state as unavailable.

Reply:
We do report the failed fetch as unavailable sensor now.
The trouble is, if the default retry is 3, some users will never get any value, they need 10 retries to get a value at all.
But users with reliable net do not want 10 retries, they want to have just 3 to report unavailable sensor asap.

Reply:
Yes, we can set the default to eg three retries and let users that need something else customize the data fetching including trying the fetch again in an automation. They can then do retry in multiples of 3 until they have a result.

Reply:
I really disliked removal of something useful, but I have removed the optional parameters as requested.
(The retry timeout default was increased to 10. It may impact users with high frequency polling but the consequences for them are less critical then those with unreliable nets who really need 10 retries.)